### PR TITLE
feat(onboarding): Add heartbeat footer to instruction's page - [TET-610]

### PR DESCRIPTION
**Pull request by priscilawebdev**
**Required Feature Flag:**

`organizations:onboarding-heartbeat-footer`

**Description**

As a Sentry user going through the project creation wizard, I want to receive clear feedback that my app is connected to this new project (Install and Configure steps), so that I know I’ve done well this far, before attempting to send my first error.

**Previews:**

_Server connection(sessions + first transaction) + First error_

https://user-images.githubusercontent.com/29228205/209944065-e505bb45-7437-4d90-b664-429f1ec962df.mov

_Server connection(first transaction) + First error_

https://user-images.githubusercontent.com/29228205/209945115-9651b282-c282-40d2-820a-2e2e7fdb16bb.mov

_First error only_

https://user-images.githubusercontent.com/29228205/209945427-27dc61f0-e411-41fe-a32c-f99561d929ea.mov

_Route Change Modal_

https://user-images.githubusercontent.com/29228205/209971872-9e9f40e6-7e07-42aa-90ba-d1850ea00f9e.mov

**Caveats**

_Sessions_

- It is necessary to add the `release` property to `Sentry.init()` to start a session and this information is not part of the code snippet. We will have to update the documents, so users will copy this part too.

- According to the [docs](https://develop.sentry.dev/sdk/sessions/#terminal-session-states):

> A session can exist in two states: in progress or terminated. A terminated session must not receive further updates. exited, crashed and abnormal are all terminal states. 

Is the 'exit' state the same as 'errored'? Is the current logic in the code correct?

**PS**

This is behind a feature flag but would be available for all platforms, since we not relay only on `session` but also on the `firstTransactionEvent` and `firstEvent(error)`

As more design changes are being worked on, tests are not planned for changes to come in this PR

### CODE REVIEW

File: static/app/views/projectInstall/heartbeatFooter/index.tsx [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+
+  const project =
+    !projectsLoading && !fetchError && projects.length ? projects[0] : undefined;
+
+  const {
+    sessionLoading,
+    eventLoading,
+    firstErrorReceived,
+    firstTransactionReceived,
+    sessionInProgress,
```
Comment (priscilawebdev) [inline static/app/views/projectInstall/heartbeatFooter/index.tsx:71]: `sessionInProgress`, `firstTransactionReceived` and  `firstErrorReceived ` fulfills all the 3 proposed options of the [Heartbeat RFC](https://github.com/getsentry/rfcs/blob/21098c90046db4247dd7046b51f1bf6ae832dfbf/text/0044-heartbeat.md#options-considered)

**Comment by evanpurkhiser** [comment]
Ooo I definitely want to take a look at this next week!

**Comment by priscilawebdev** [comment]
**UPDATES:**

The beats + platform icon + name will no longer be shown on mobile platforms. The new changes are in the PR https://github.com/getsentry/sentry/pull/42704

image

**Comment by priscilawebdev** [comment]
_Updated Check Mark_

image

File: static/app/views/projectInstall/heartbeatFooter/useHeartbeat.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  const {isLoading: sessionLoading} = useQuery<SessionApiResponse>(
+    [
+      `/organizations/${organization.slug}/sessions/`,
+      {
+        query: {
+          project: project?.id,
+          groupBy: 'session.status',
+          statsPeriod: '24h',
+          interval: '1h',
+          field: [SessionFieldWithOperation.SESSIONS],
```
Comment (matejminar) [inline static/app/views/projectInstall/heartbeatFooter/useHeartbeat.tsx:50]: 1. Do we need to group by session.status here? The existence of any session kind(status) proves that the SDK is connected, no? We could greatly simplify the onSuccess handler (just check the total sessions count).
2. Is there a need for an interval of 1h when we have statsPeriod 24h? I would keep both statsPeriod and interval at 1d (or even 1h).

Comment (priscilawebdev) [reply static/app/views/projectInstall/heartbeatFooter/useHeartbeat.tsx]: Thanks for the great feedback @matejminar 🌮 

1. Updated! 
2. When I use 1d/1h for both interval and statsPeriod, I don't get a fresh session. Do we need to set both? How important is it to set the interval? How does 24h differ from 1d?

Comment (priscilawebdev) [reply static/app/views/projectInstall/heartbeatFooter/useHeartbeat.tsx]: leaving only statsPeriod with 24h brings me recent sessions

**Review by matejminar — APPROVED** [review]

**Pull request merged**